### PR TITLE
[Core][ModelPartIO/CodeLocation] clean whitespace function

### PR DESCRIPTION
**Pull request by philbucher**
- replacing custom implementation for standard function `std::isspace`
- Improved a fct in the `ModelPartIO`

### CODE REVIEW

File: kratos/sources/model_part_io.cpp [hunk 1]
```diff
@@ -5145,20 +5140,9 @@ char ModelPartIO::GetCharacter() //Getting the next character skipping comments
...
-    {
-        std::stringstream buffer;
-        buffer << "A \"" << rStatement << "\" statement was expected but the given statement was \"";
-        buffer <<  rGivenWord << "\"" << " [Line " << mNumberOfLines << " ]";
-        KRATOS_ERROR << buffer.str() << std::endl;
-    }
-    else
-        result = true;
-
-    return result;
```
Comment (philbucher) [inline kratos/sources/model_part_io.cpp:5161]: what a beauty :sweat_smile:

File: kratos/sources/model_part_io.cpp [hunk 2]
```diff
@@ -1123,9 +1123,9 @@ void ModelPartIO::SkipBlock(std::string const& BlockName)
         if(word == "End")
         {
             ReadWord(word);
-            if(number_of_nested_blocks == 0){
-                    if(CheckStatement(word , BlockName))
```
Comment (philbucher) [inline kratos/sources/model_part_io.cpp:1127]: this function always returned `true`, or it threw

**Review by matekelemen — COMMENTED** [review]
👌

File: kratos/includes/model_part_io.h [hunk 3]
```diff
@@ -713,11 +713,9 @@ class KRATOS_API(KRATOS_CORE) ModelPartIO : public IO
 
     char SkipWhiteSpaces();
 
-    bool IsWhiteSpace(char C);
-
     char GetCharacter();
 
-    bool CheckStatement(std::string const& rStatement, std::string const& rGivenWord);
```
Comment (matekelemen) [inline kratos/includes/model_part_io.h:720]: this function declaration, along with the rest of the `private` crap only used inside this class, could be just moved to source as well.

Comment (philbucher) [reply kratos/includes/model_part_io.h:720]: fully agree

You ok if I do it in a separate PR?

Comment (matekelemen) [reply kratos/includes/model_part_io.h:720]: of course

**Review by matekelemen — APPROVED** [review]

**Pull request merged**